sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/types/Arguments.kt

```diff
@@ -74,8 +74,7 @@ fun List<Argument>.toSqliteArguments(): List<Argument> {
   val numbersToSkip = LinkedHashSet<Int>()
   val nameAllocator = NameAllocator()
   var highestNumber = 0
-
-  distinctBy { it.ranges }.sortedBy { it.ranges[0].start }.forEachIndexed { i, original ->
+  distinctBy { it.ranges }.sortedBy { it.ranges[0].start }.run { forEachIndexed { i, original ->
```

Comment:
`get` calls inside `forEachIndexed` were referencing the original list instead of the sorted list causing some rando bugs.